scripts/postbuild.js

```diff
@@ -0,0 +1,59 @@
+const path = require('node:path');
+const fs = require('node:fs/promises');
+
+// Copy /myPath.html to /myPath/index.html
+// This ensures both URL patterns work: /myPath and /myPath/
+async function generateIndexHtmlFiles(outDir) {
+	console.log('Post-build: Creating index.html files from hoisted pages...');
+
+	// Walk through all directories recursively
+	async function* walkDirs(dir) {
+		const dirents = await fs.readdir(dir, { withFileTypes: true });
```

Comment:
https://nodejs.org/docs/latest/api/fs.html#fspromisesreaddirpath-options

FWIW you could use the `recursive` option for `fs.readdir`